pygments/lexers/markup.py

```diff
@@ -571,13 +571,17 @@ def _handle_codeblock(self, match):
             # warning: the following rules eat outer tags.
             # eg. **foo _bar_ baz** => foo and baz are not recognized as bold
             # bold fenced by '**'
-            (r'([^\*]?)(\*\*[^* \n][^*\n]*\*\*)', bygroups(Text, Generic.Strong)),
+            (r'(?<!\*)(\*\*[^* \n]([^*\n]+\n?)*(?<![ \n])\*\*)(?!\*)',
```

Comment:
`bygroups` is no longer necessary.

More importantly, this regex (as well as the others below) has catastrophic backtracking. Try

```python
>>> import re
>>> re.match(r'(?<!\*)(\*\*[^* \n]([^*\n]+\n?)*(?<![ \n])\*\*)(?!\*)', '**' + 'a'*100) # hangs
```

See `Contributing.md` for explanations and remedies.